stan/math/opencl/kernels/matrix_multiply.hpp

```diff
@@ -79,13 +79,13 @@ static const char* matrix_multiply_kernel_code = STRINGIFY(
       // is 0 and the end tile is num_tiles-1 which
       // is then a general matrix multiply
       const int end_tile_A
-          = lower_upper_A == LOWER ? (i / THREAD_BLOCK_SIZE) : (num_tiles - 1);
+          = lower_upper_A & UPPER ? (num_tiles - 1) : (i / THREAD_BLOCK_SIZE);
```

Comment:
Is the performance gain of this non-negligible? If not, we should not use this, otherwise we need to somehow prevent developers from changing enum values.

Reply:
Did not check it, but I guess it is negligible. How would you prevent changes? Just add a coment that the values are coulpled to bitwise operations?